sensor/common/registry/registry_store.go

```diff
@@ -99,6 +100,16 @@ func (rs *Store) UpsertRegistry(ctx context.Context, namespace, registry string,
 	return nil
 }
 
+// UpsertNoAuthRegistry will store a new registry with no user/pass and return the created registry integration
+func (rs *Store) UpsertNoAuthRegistry(ctx context.Context, namespace string, imgName *storage.ImageName) (registryTypes.Registry, error) {
```

Comment:
Sounds good, I modified the signature to `UpsertRegistry` - which had a slight cascade effect.  In the end believe this will be more efficient since the Upsert no longer has to 'search' for the registry it just created and can just return it